web/telemetry/host_telemetry.go

```diff
@@ -0,0 +1,61 @@
+package telemetry
+
+import (
+	"errors"
+	"time"
+
+	"github.com/trento-project/trento/web/models"
+	"gorm.io/gorm"
+)
+
+type HostTelemetry struct {
+	InstallationID string    `json:"installation_id"`
+	AgentID        string    `json:"agent_id"`
+	SLESVersion    string    `json:"sles_version"`
+	CPUCount       int       `json:"cpu_count"`
+	SocketCount    int       `json:"socket_count"`
+	TotalMemoryMB  int       `json:"total_memory_mb"`
+	CloudProvider  string    `json:"cloud_provider"`
+	Time           time.Time `json:"time"`
+}
+
+type HostTelemetries []*HostTelemetry
+
+type HostTelemetryExtractor struct {
+	installationIdAwareExtractor
+	db *gorm.DB
+}
+
+func (ex *HostTelemetryExtractor) Extract() (interface{}, error) {
+	var collectedHostsTelemetry []models.HostTelemetry
```

Comment:
I'm a bit confused with `models/entities` here. This `models.HostTelemetry` should be an entity in reality, right?
As it loads the data from the database, that later on should be transformed to a model (which in this case is the publishing data model).

Reply:
mh :thinking: I understand the feeling

`models.HostTelemetry` is what gets projected from the discoveries, that's why it went in `models`.
That package started as containing ReadModels.

I feel the need to clarify what we consider an `entity` and what a `model`/`readmodel` is
we started introducing `models` which we initially reasoned a bout as read models, then we introduced also entities.

I feel this is important to clarify ASAP as it defines the application layering as well and thus future architectural improvements.

Very shortly what I see is:
- `entities`: represent somehow the DomainModel, they have their own lifecycle, business logic and are stored somewhere. Possibly write Only (this might evolve in Aggregates, but not now)
- `readmodels`: the result of different things happening in the system, represents usually a use case (HostOverview, HostDetail, ClusterOverview...). In an upstreamdownstream relashionship usually the downstream defines its own readmodels as a result of things happening in the system ([@ziobrando](https://twitter.com/ziobrando) says it). A read model might or might not be the same representation that goes in the wire to the frontend let's say. In this case we could reason about `ViewModels` for instance...

It's a complex topic and I'd love to go deeper, refine and clarify as a team what they mean and when/how their supposed to be, not sure this is the right moment, though.

For this specific comment I'd be ok in moving `models.HostTelemetry` to `entities.HostTelemetry`, nothign against it.
@fabriziosestito @dottorblaster any thoughts on this?

Reply:
Moved `models.HostTelemetry` to `entities.HostTelemetry`.
Eitherway it's the same for now. :stuck_out_tongue_closed_eyes: